xarray/core/utils.py

```diff
@@ -30,7 +31,12 @@
 
 import numpy as np
 import pandas as pd
-from typing_extensions import TypeGuard
+
+if sys.version_info >= (3, 10):
+    from typing import TypeGuard
+else:
+    from typing_extensions import TypeGuard
```

Comment:
It should be possible to import `TypeGuard` from `typing` with Python 3.10:  https://www.python.org/dev/peps/pep-0647/

Reply:
not sure if I'm missing something but I think it would be better to catch the raised `ImportError`:
```suggestion
try:
    from typing import TypeGuard
except ImportError:
    from typing_extensions import TypeGuard
```

Reply:
Actually, I restored the `sys.version_info >= (3, 10)` implementation because mypy is not happy with the try/except...